appinventor/components/src/com/google/appinventor/components/runtime/FusiontablesControl.java

```diff
@@ -404,16 +407,50 @@ public void ForgetLogin() {
     OAuth2Helper.resetAccountCredential(activity);
   }
 
+  //InsertRow receives a String or a list as a value
   @SimpleFunction(
     description="Inserts a row into the specified fusion table. The tableId field is the id of the" +
       "fusion table. The columns is a comma-separated list of the columns to insert values into. The" +
       " values field specifies what values to insert into each column.")
-  public void InsertRow(String tableId, String columns, String values) {
-    query = "INSERT INTO " + tableId + " (" + columns + ")" + " VALUES " + "(" + values + ")";
+  public void InsertRow(String tableId, String columns, Object values) {
+    String reformattedValues = "";
+    if(values != null && values instanceof String || values != null && values instanceof gnu.lists.FString) {
+      Log.i("FusionTablesInsertRow", "String:"+values);
+      String valuesString = values.toString();
+      //Single quotes are escape chars and must be replaced with two single quotes (Natalie)
+      if(valuesString.startsWith("\'")){
+        reformattedValues = valuesString;
+      }
+      else if(valuesString.startsWith("\"")) {
+        String quotifiedValues = valuesString.replace("\'", "\'\'");
+        reformattedValues = quotifiedValues.replace("\"", "\'");
+      }
+      else{
+        //Not sure if we want to have a specific error message for this case.
+        //The code below is also probably an incorrect implementation of error messages (Natalie)
+        /*form.dispatchErrorOccurredEventDialog(this, "SendQueryValuesFormat",
```

Comment:
I think it would be good to leave this error reporting in or dispatch it to the component's error event. Otherwise, the component will do nothing and the reason will be non-obvious.